examples/medplum-chart-demo/src/components/graphs/ObservationGraph.tsx

```diff
@@ -0,0 +1,56 @@
+import { Paper } from '@mantine/core';
+import { formatDate, getReferenceString } from '@medplum/core';
+import { Coding, ObservationComponent, Patient } from '@medplum/fhirtypes';
+import { useMedplum } from '@medplum/react';
+import { useEffect, useState } from 'react';
+import { LineChart } from './LineChart';
+import { measurementsMeta } from './MeasurementData';
+import { ChartData, ChartDataset } from 'chart.js';
+
+interface ObservationGraphProps {
+  code: Coding;
+  patient: Patient;
+}
+
+export function ObservationGraph(props: ObservationGraphProps): JSX.Element {
+  const medplum = useMedplum();
+
+  const { chartDatasets } = measurementsMeta[props.code.display as string];
+  const [chartData, setChartData] = useState<ChartData<'line', number[], string>>();
+  const observations = medplum
+    .searchResources('Observation', {
+      code: props.code.code,
+      patient: getReferenceString(props.patient),
+    })
+    .read();
+
+  useEffect(() => {
+    if (observations) {
+      const labels: string[] = [];
+      const datasets: ChartDataset<'line', number[]>[] = chartDatasets.map((item) => ({ ...item, data: [] }));
+
+      for (const observation of observations) {
```

Comment:
Might make sense to break this into a helper function that just converts a list of observations into a list of numbers